roles/kubernetes_setup/tasks/main.yml

```diff
@@ -33,6 +33,35 @@
     - "{{ kube_controller_manager }}"
     - "{{ kube_scheduler }}"
 
+- name: Define kube 1.7 master nodes images
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  set_fact:
+    kube_maj_ver: "1.7"
+    kube_apiserver: "registry.fedoraproject.org/f27/kubernetes-apiserver"
+    kube_controller_manager: "registry.fedoraproject.org/f27/kubernetes-controller-manager"
+    kube_scheduler: "registry.fedoraproject.org/f27/kubernetes-scheduler"
+
+- name: Define kube 1.7 worker nodes and etcd images
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  set_fact:
+    kube_maj_ver: "1.7"
+    kube_kubelet: "registry.fedoraproject.org/f27/kubernetes-kubelet"
+    kube_proxy: "registry.fedoraproject.org/f27/kubernetes-proxy"
+    etcd: "registry.fedoraproject.org/f27/etcd"
+
+
+- name: Install master node system containers that is, k8-apiserver, k8-controller_manager and k8-scheduler
+  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
+  command: atomic install --system --system-package=no --name kube-apiserver {{ kube_apiserver }}
+  command: atomic install --system --system-package=no --name kube-controller-manager {{ kube_controller_manager }}
+  command: atomic install --system --system-package=no --name kube-scheduler {{ kube_scheduler }}
+
```

Comment:
I think these commands have to be in separate tasks or put into a loop.  Only one command will run (at least that is how it is on Ansible 2.2.x).

```
- name: Install master node system containers that is, k8-apiserver, k8-controller_manager and k8-scheduler
  when: ansible_distribution == "Fedora" and ansible_distribution_major_version > '26'
￼  command: {{ item }}
  with_items:
    - atomic install --system --system-package=no --name kube-apiserver {{ kube_apiserver }}
￼    - atomic install --system --system-package=no --name kube-controller-manager {{ kube_controller_manager }}
￼    - atomic install --system --system-package=no --name kube-scheduler {{ kube_scheduler }}
```

Reply:
@mike-nguyen  Thanks, I was so much looking at k8 and other things, I missed this totally.